app/backend/forum/views.py

```diff
@@ -83,82 +96,33 @@ def get_all_posts(request):
     if page <= 0:
         page = 1
     query += "offset " + str((page-1)*page_size) + " limit " + str(page_size) + " "
-    authors = []
 
-    if search_query:
+    try:
+        user = CustomUser.objects.get(email = request.user.email)
+        followed = user.followed_categories
+        followed_categories = set()
+        for i in followed:
+            followed_categories.add(i)
 
-        try:
-            doctors = Doctor.objects.filter(full_name__icontains=search_query)
-            for doctor in doctors:
-                author = doctor.user
-                authors.append(author)
-        except:
-            pass
-        try:
-            members = Member.objects.filter(member_username__icontains=search_query)
-            print(members)
-            for member in members:
-                print(member.member_username)
-                author = member.user
-                authors.append(author)
-        except:
-            pass
+    except:
+        user = None
+        followed_categories = set()
 
-    posts_by_user = Post.objects.filter(author__in=authors)
-    post_objects = Post.objects.raw(query.format("distinct *"))
+    followed_categories = tuple(followed_categories)
 
 
 
-    try:
-        user = CustomUser.objects.get(email = request.user.email)
-    except:
-        user = None
-    
-    posts = []
-    for post in posts_by_user:
-        serializer_post_data = PostSerializer(post).data
-        if user:
-            if post.id in user.upvoted_posts:
-                serializer_post_data['vote'] = 'upvote'
-            elif post.id in user.downvoted_posts:
-                serializer_post_data['vote'] = 'downvote'
-            else:
-                serializer_post_data['vote'] = None
 
-            if post.id in user.bookmarked_posts:
-                serializer_post_data['bookmark'] = True
-            else:
-                serializer_post_data['bookmark'] = False
-        else:
-            serializer_post_data['vote'] = None
-            serializer_post_data['bookmark'] = None
+    query = query.format(f"distinct forum_post.id, date, (CASE WHEN(category_id IN {followed_categories}) THEN 1 ELSE 0 END) as c1")
+    new_query = f"SELECT * FROM ({query}) AS A1 ORDER BY A1.c1 desc"
```

Comment:
Less readable, and a lot of changes. Comments to help understand easier would have been nice.